include/revng/Model/VerifyHelper.h

```diff
@@ -59,6 +61,12 @@ class VerifyHelper {
   }
 
 public:
+  void evict(const model::Type *T) {
```

Comment:
The commit message says:
> ...a performance hit that cannot be taken...

Is it actually the case?
Such an API induces the user to try and keep up-to-date an inherently temporary data structure, which is a delicate thing to do. Let's do this only if necessary, otherwise let's start over with a new `VerifyHelper` and be done with it.

Reply:
I'm measuring the performance penalty of avoiding this in the only place I need it. Will report back. I'd be happy to avoid introducing this.

Reply:
Cool I managed to drop this.